src/level_set/FESurfaceDistanceEvaluator.cpp

```diff
@@ -1254,23 +1254,22 @@ FESurfaceDistanceEvaluator::computeSignedDistanceSurfaceMesh(int n_idx, int d_id
                     avg_proj += proj;
 
                     // Get the nodes
-                    const libMesh::Point& n0 = elem->point(0);
-                    const libMesh::Point& n1 = elem->point(1);
-                    X_node[0][0] = n0(0);
-                    X_node[0][1] = n0(1);
-                    X_node[1][0] = n1(0);
-                    X_node[1][1] = n1(1);
-
-#if (NDIM == 3)
-                    const libMesh::Point& n2 = elem->point(2);
-                    X_node[2][0] = n2(0);
-                    X_node[2][1] = n2(1);
-                    X_node[2][2] = n2(2);
+                    const unsigned int n_nodes = elem->n_nodes();
+#if !defined(NDEBUG)
+                    TBOX_ASSERT(n_nodes == NDIM);
 #endif
```

Comment:
This doesn't seem right. Do we really only support TRI3 and EDGE2?

Reply:
Yes. We check this condition  in the constructor and map intersection routine.

The computational geometry work done in this class is quite expensive to carry out. It has been optimized to do it for EDGE2 and TRI3, the two most common elements for engineering geometries.